### PR TITLE
add blender package

### PR DESCRIPTION
**Pull request by ppodhajski**

### CODE REVIEW

File: var/spack/repos/builtin/packages/blender/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            src = os.path.join(self.stage.source_path, file)
+            dst = os.path.join(prefix, file)
+            if os.path.isdir(src):
+                shutil.copytree(src, dst, False)
+            else:
+                shutil.copy2(src, dst)
+
+    def setup_environment(self, spack_env, run_env):
+        blender_python_path = os.path.join(prefix, self.version.string[:4], 'python/lib/python3.7')
+        run_env.set('PYTHONPATH', blender_python_path)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/blender/package.py:32]: `depends_on("python")`? Or does it ship Python? In the latter case, it should conflict…

File: var/spack/repos/builtin/packages/blender/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def url_for_version(self, version):
+	glibc_version = version.string.split('-')[1]
+	blender_version_full = version.string.split('-')[0]
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/blender/package.py]:
```suggestion
	(glibc_version, blender_version_full) = version.string.split('-')
```

File: var/spack/repos/builtin/packages/blender/package.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	blender_version_minor = version.string[:4]
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/blender/package.py]: `version.upto(2)` does not work?

Comment (ppodhajski) [reply var/spack/repos/builtin/packages/blender/package.py]: Error: AttributeError: 'Version' object has no attribute 'upto'

Comment (matz-e) [reply var/spack/repos/builtin/packages/blender/package.py]: https://spack.readthedocs.io/en/latest/spack.html#spack.version.Version.up_to

Forgot an underscore…

Comment (matz-e) [reply var/spack/repos/builtin/packages/blender/package.py]: It would be better than relying on indices only. What about blender `2.100` or `3.1`?

**Review by matz-e — CHANGES_REQUESTED** [review]
See comments.

File: var/spack/repos/builtin/packages/blender/package.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    version('2.81a-217', sha256='bb6e03ef79d2d7273336f8cfcd5a3b3f')
+
+    conflicts("python")
+
+    def install(self, spec, prefix):
+	for file in os.listdir(self.stage.source_path):
+            src = os.path.join(self.stage.source_path, file)
+            dst = os.path.join(prefix, file)
+            if os.path.isdir(src):
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/blender/package.py:25]: This seems to be binary release? In that case, can we add conflict for non-supported platforms?

**Review by pramodk — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/blender/package.py (hunk 4 above)
Comment (ppodhajski) [reply var/spack/repos/builtin/packages/blender/package.py:25]: Yes, it is a binary release. 
The only conflict specyfing targets I found it the repo are `ppc64`  and `ppc64le`. Is it enough?

Comment (pramodk) [reply var/spack/repos/builtin/packages/blender/package.py:25]: this is only for Linux so we need to add conflict for OS X and PowerPC

**Review by matz-e — APPROVED** [review]

**Review by pramodk — APPROVED** [review]

**Comment by matz-e** [comment]
delete branch?

File: var/spack/repos/builtin/packages/blender/package.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    version('2.81a-217', sha256='bb6e03ef79d2d7273336f8cfcd5a3b3f')
+
+    conflicts("python")
+    conflicts('platform=darwin', msg='this binary release of blender requires Linux')
+    conflicts('platform=ppc64', msg='this binary release of blender requires Linux')
+
+
+    def install(self, spec, prefix):
+	for file in os.listdir(self.stage.source_path):
```
Comment (ferdonline) [inline var/spack/repos/builtin/packages/blender/package.py:22]: ??
==> Error: invalid syntax in /Users/leite/dev/spack/var/spack/repos/builtin/packages/blender/package.py, line 23

/Users/leite/dev/spack/var/spack/repos/builtin/packages/blender/package.py:22:48: E999 TabError: inconsistent use of tabs and spaces in indentation

**Pull request merged**